src/blueapi/worker/task.py

```diff
@@ -22,7 +22,7 @@ class Task(BlueapiBaseModel):
 
     def prepare_params(self, ctx: BlueskyContext) -> Mapping[str, Any]:
         model = _lookup_params(ctx, self)
-        return _model_to_kwargs(model)
+        return model.model_dump(warnings=False)
```

Comment:
This might need to be `{field: getattr(model, field) for field in model.model_dump(warnings=False)}`.

@callumforrester ,  do you have an example of the nested model you wanted to avoid in the `_model_to_kwargs` method?